packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -568,6 +583,21 @@ Future<void> diagnoseXcodeBuildFailure(XcodeBuildResult result, Usage flutterUsa
   }
 }
 
+enum XcodeBuildAction { build, archive }
```

Comment:
nit, document a little bit

Reply:
also, if you never switch over this consider using String constants

Reply:
I prefer comparing enums to strings whether it's in a switch or an if conditional, strings are for human readability (and for wrapping a bunch of shelled out commands, as we do...). It also adds some pseudo type checking when being passed around.  And not that it matters in this context, but the comparison is faster (at least in most languages, maybe dart does something as fast with a const string?).  The only time when you have to think about it a bit is when you're serializing it to make sure if you insert one into the middle of the list it doesn't change the order of the enumeration symbols.  But I'd rather serialize an int than a string in that case anyway.

https://github.com/dart-lang/language/issues/158 would be nice.

Reply:
I don't really think the performance will matter for this code, but ack on the usability.